File: flutter-sdk/lib/src/domain/models/chat_message/cta_button.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:equatable/equatable.dart';

// A button with text and URL used in CTA messages
class CTAButton extends Equatable {
  final String text;
  final String url;

  const CTAButton({required this.text, required this.url});

  @override
  List<Object?> get props => [text, url];
}

File: android-sdk/sdk/src/commonMain/kotlin/com/yalo/chat/sdk/domain/model/CtaButton.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.domain.model

import kotlinx.serialization.Serializable

// A CTA button has a display label and a URL to open when tapped.
@Serializable
data class CtaButton(
    val text: String,
    val url: String,
)
